sklearn/ensemble/gradient_boosting.py

```diff
@@ -850,25 +850,26 @@ def _check_params(self):
             if self.max_features == "auto":
                 # if is_classification
                 if self.n_classes_ > 1:
-                    max_features = max(1, int(np.sqrt(self.n_features)))
+                    max_features = max(1, int(np.sqrt(self._n_features)))
```

Comment:
I'm not an expert of that code but it is important to keep `n_features` in memory.
Can we change the signature of `_check_params(self)` -> `_check_params(self, X, y)` ?

`X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'], dtype=DTYPE)` can also be done in `_check_params`.